src/command_line_interface/start_interface.rs

```diff
@@ -0,0 +1,64 @@
+use crate::command_line_interface::arguments::{
+    CommandType, ListInterfaceSubcommand, RtpeekerArgs,
+};
+use crate::command_line_interface::list_devices::list_devices;
+use crate::command_line_interface::start_interface::Action::{AnalyzeFile, CapturePackets};
+use clap::Parser;
+use std::net::SocketAddr;
+use std::process::exit;
+
+pub enum Action {
+    CapturePackets(String, SocketAddr),
+    AnalyzeFile(String, SocketAddr),
+}
+
+pub fn start_command_line_interface() -> Action {
+    match RtpeekerArgs::parse().action {
+        CommandType::File(args) => {
+            if let Ok(socket_addr) = args.socket_to_serve.parse() {
+                AnalyzeFile(args.file_path, socket_addr)
+            } else {
+                parsing_socket_error();
+                exit(1);
+            }
+        }
+        CommandType::Interface(args) => {
+            if let Ok(socket_addr) = args.socket_to_serve.parse() {
+                CapturePackets(args.interface_name, socket_addr)
+            } else {
+                parsing_socket_error();
+                exit(1);
+            }
+        }
+        CommandType::ListInterfaces(sub_command) => {
+            let flags = get_flags(sub_command);
+            list_devices(flags);
+            exit(0);
+        }
+    }
+}
```

Comment:
See [this method in Trunk](https://github.com/thedodd/trunk/blob/6a94c6b6df35d1a3618307afcfac69c66a6ae939/src/main.rs#L64). Lets implement the `run` method on all of the `Parser` or `Args` structs similarly to what's done in the link above.

Reply:
Also, `clap` can take care of validation and produce nice messages when arguments are invalid, see [this](https://docs.rs/clap/latest/clap/struct.Arg.html#method.value_parser).